arc_solvers/models/qa/multi_choice/qa_multi_choice_max_att.py

```diff
@@ -0,0 +1,231 @@
+        If provided, will be used to calculate the regularization penalty during training.
+    share_encoders : ``bool``, optional (default=``false``)
+        Shares the weights of the question and choice encoders.
+    aggregate_question : ``str``, optional (default=``max``, allowed options [max, avg, sum, last])
+        The aggregation method for the encoded question.
+    aggregate_choice : ``str``, optional (default=``max``, allowed options [max, avg, sum, last])
+        The aggregation method for the encoded choice.
+    """
+
+    def __init__(self, vocab: Vocabulary,
+                 text_field_embedder: TextFieldEmbedder,
+                 question_encoder: Optional[Seq2SeqEncoder] = None,
+                 choice_encoder: Optional[Seq2SeqEncoder] = None,
+                 initializer: InitializerApplicator = InitializerApplicator(),
+                 aggregate_question: Optional[str] = "max",
+                 aggregate_choice: Optional[str] = "max",
+                 embeddings_dropout_value: Optional[float] = 0.0,
+                 params=Params) -> None:
+        super(QAMultiChoiceMaxAttention, self).__init__(vocab)
+
+        self._use_cuda = (torch.cuda.is_available() and torch.cuda.current_device() >= 0)
+
+        self._params = params
+
+        self._text_field_embedder = text_field_embedder
+        if embeddings_dropout_value > 0.0:
+            self._embeddings_dropout = torch.nn.Dropout(p=embeddings_dropout_value)
+        else:
+            self._embeddings_dropout = lambda x: x
+
+        self._question_encoder = question_encoder
+
+        # choices encoding
+        self._choice_encoder = choice_encoder
+
+        self._question_aggregate = aggregate_question
+        self._choice_aggregate = aggregate_choice
+
+        self._num_labels = vocab.get_vocab_size(namespace="labels")
+
+        question_output_dim = self._text_field_embedder.get_output_dim()
+        if self._question_encoder is not None:
+            question_output_dim = self._question_encoder.get_output_dim()
+
+        choice_output_dim = self._text_field_embedder.get_output_dim()
+        if self._choice_encoder is not None:
+            choice_output_dim = self._choice_encoder.get_output_dim()
+
+        if question_output_dim != choice_output_dim:
+            raise ConfigurationError("Output dimension of the question_encoder (dim: {}), "
+                                     "plus choice_encoder (dim: {})"
+                                     "must match! "
+                                     .format(question_output_dim,
+                                             choice_output_dim))
+
+        # question to choice attention
+        att_question_to_choice_params = params.get("att_question_to_choice")
+        if "tensor_1_dim" in att_question_to_choice_params:
+            # automatically infer the input size
+            att_question_to_choice_params = update_params(att_question_to_choice_params,
```

Comment:
See above - try to construct this Similarity Function directly, passing in options which allow you to do that, rather than passing in a `Params` object.

Reply:
Good point. I couldn't find the documentation of the semantics of the Options and how to get options from Params.

Reply:
I'm not quite sure what you mean - If you look at any of the `SimilarityFunctions`, you should be able to see what args they take:

e.g. https://github.com/allenai/allennlp/blob/v0.4.0/allennlp/modules/similarity_functions/linear.py#L71

So you could build that in json like:

```
similarity : {
  "type": "linear",
  "tensor_1_dim" ...
}
```

So in the same way you are constructing and passing in a `Seq2SeqEncoder` from your config file, you would do the same here.